### DISCUSSION TITLE
How to print the weights in forward pass of each layer and gradient loss and weight update in backward propagation?

### DISCUSSION THREAD

**Discussion by utkarsh0902311047** [pyg-team/pytorch_geometric#7597] [Q&A]
Hello,
Please guide me how can I print the weights starting from initialization to weights of each layer and then gradient loss and weight update in backward propagation? And how it can be printed for each epoch?

Thank you

**Comment by wsad1** [comment]
depending on which layer you are using the `.weight` parameter should give you the weight of the model, and `weight.grad` the gradient. This [answer](https://discuss.pytorch.org/t/how-to-print-the-computed-gradient-values-for-a-network/34179/3) might help.
